pilot/pkg/ambient/ambient.go

```diff
@@ -17,13 +17,16 @@ package ambient
 type NodeType = string
 
 const (
-	LabelStatus = "istio.io/ambient-status"
-	TypeEnabled = "enabled"
-	// LabelType == "workload" -> intercept into ztunnel
-	// TODO this could be an annotation – eventually move it into api repo
-	LabelType = "ambient-type"
-
-	TypeWorkload NodeType = "workload"
-	TypeNone     NodeType = "none"
+	// AnnotationType determines the type of workload this is.
+	AnnotationType = "ambient.istio.io/type"
+	// TypeMesh indicates this workload is in the mesh and has redirection to ztunnel.
+	// This is added by the CNI when redirection is configured, not by the user.
+	TypeMesh NodeType = "mesh"
+	// TypeDisabled indicates this workload does NOT have redirection enabled. Workloads without any
+	// value should be assumed to be "disabled" as well, but explicitly setting this acts as an opt-out.
+	TypeDisabled NodeType = "disabled"
+	// TypeZtunnel indicates this workload is a ztunnel.
+	TypeZtunnel NodeType = "ztunnel"
+	// TypeWaypoint indicates this workload is a waypoint.
 	TypeWaypoint NodeType = "waypoint"
```

Comment:
```suggestion
	TypeZtunnel NodeType = "ztunnel"
	// TypeWaypoint indicates this workload is a waypoint.
	TypeWaypoint NodeType = "waypoint"
```

If this label is purely informational now, and doesn't "do" anything now, aren't states that aren't `mesh` or `<unset>` largely redundant?

i.e. presumably CNI is using $some-other-annotation to decide whether something is a waypoint or ztunnel, and not this - so what's the point of this?

Reply:
We could put `ambient.istio.io/type: disabled` on waypoint and ztunnel, but it seems a bit confusing to say they are "disabled". Maybe if we rename the annotation to `ambient.istio.io/redirection: {enabled,disabled}`?

Reply:
Yeah - I kinda like that better. I agree making this an informational flag that reflects a decision the CNI has already made is a good improvement, and if that's the case really a two-state informational flag.

Reply:
k will update

Reply:
That could work if the only purpose is for redirection.  I had thought there is other purpose.

Reply:
It is only to signal if redirection is enabled. We need a way to signal to things if there is a ztunnel so we know whether to use HBONE basically

Reply:
In fact, we had people don't understand if ingress gateway or waypoint are captured by ztunnel so this would bring some clarity too!